examples/demo.rs

```diff
@@ -1,11 +1,13 @@
 extern crate capstone;
 
+use capstone::*;
+
 const CODE: &'static [u8] = b"\x55\x48\x8b\x05\xb8\x13\x00\x00";
 
-fn example() -> capstone::CsResult<()> {
-    let mut cs = capstone::Capstone::new(capstone::Arch::X86, capstone::Mode::Mode64)?;
-    cs.set_detail(true).unwrap();
-    cs.att();
+fn example() -> CsResult<()> {
+    let mut cs = Capstone::new().x86().mode(x86::ArchMode::Mode64).build()?;
+    cs.set_detail(true)?;
+    cs.set_syntax(Syntax::Att)?;
```

Comment:
why isn't detail in the builder?

Similarly, syntax; for same reason if you set the mode to `x86::ArchMode::Mode64` on `.arm()` its a failure, i think syntax makes sense too, just from ergonomics perspective.

It's like I create a builder; then build some more (which can all fail)

Reply:
I suppose I can add `detail()` to the builder. I originally did not to keep the code simpler, but at this point it doesn't matter. Should not be much code to add.

You actually can call `.syntax()`--I just need change the example. I agree that the builder is nice because code will fail to compile if you pass a nonsensical value to the builder's methods. We still have the `new_raw()` method if you *really* don't want to use a builder.